/***************************************************************************
 *   Copyright (C) 2020 by Santiago GonzÃ¡lez                               *
 *                                                                         *
 ***( see copyright.txt file at root folder )*******************************/

#include "editorprop.h"
#include "codeeditor.h"
#include "basedebugger.h"

EditorProp::EditorProp( QWidget* parent )
          : QDialog( parent )
{
    setupUi(this);

    fontSize->setValue( CodeEditor::fontSize() );
    tabSize->setValue( CodeEditor::tabSize() );
    tabSpaces->setChecked( CodeEditor::spaceTabs() );
    showSpaces->setChecked( CodeEditor::showSpaces() );
}

void EditorProp::on_fontSize_valueChanged( int size )
{
    CodeEditor::setFontSize( size );
}

void EditorProp::on_tabSize_valueChanged( int size )
{
    CodeEditor::setTabSize( size );
}

void EditorProp::on_tabSpaces_toggled( bool spaces )
{
    CodeEditor::setSpaceTabs( spaces );
}

void EditorProp::on_showSpaces_toggled( bool show )
{
    CodeEditor::setShowSpaces( show );
}

